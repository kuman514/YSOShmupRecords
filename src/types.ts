export interface NavNodeInfo {
  id: string;
  label: string;
  childNavNodeIds: NavNodeInfo['id'][];
}

export interface ShmupRecord {
  id: string;
  when: Date;
  subjectId: string;
  stage: string;
  /**
   * Why score should be string?
   * In case of exceeding `Number.MAX_SAFE_INTEGER`, which equals to `9007199254740991`.
   */
  score: string;
  byWhat: string;
  comment: string;
  thumbnailUrl: string;
  originalImageUrl: string;
  tweetUrl: string;
  youtubeUrl?: string;
  specialTags?: string[];
}
