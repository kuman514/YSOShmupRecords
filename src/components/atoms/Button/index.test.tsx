import React from 'react';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';

import { Button } from '.';
import { ButtonType } from '^/types';

describe('Sample test', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match for solid button', () => {
    const { container } = render(
      <Button type={ButtonType.SOLID} isDisabled={false} onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for line button', () => {
    const { container } = render(
      <Button type={ButtonType.LINE} isDisabled={false} onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for round solid button', () => {
    const { container } = render(
      <Button
        type={ButtonType.ROUND_SOLID}
        isDisabled={false}
        onClick={() => {}}
      >
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for round line button', () => {
    const { container } = render(
      <Button
        type={ButtonType.ROUND_LINE}
        isDisabled={false}
        onClick={() => {}}
      >
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for clear button', () => {
    const { container } = render(
      <Button type={ButtonType.CLEAR} isDisabled={false} onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for disabled solid button', () => {
    const { container } = render(
      <Button type={ButtonType.SOLID} isDisabled onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for disabled line button', () => {
    const { container } = render(
      <Button type={ButtonType.LINE} isDisabled onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for disabled round solid button', () => {
    const { container } = render(
      <Button type={ButtonType.ROUND_SOLID} isDisabled onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for disabled round line button', () => {
    const { container } = render(
      <Button type={ButtonType.ROUND_LINE} isDisabled onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for disabled clear button', () => {
    const { container } = render(
      <Button type={ButtonType.CLEAR} isDisabled onClick={() => {}}>
        kuman514
      </Button>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be clickable', () => {
    const mockFn = vi.fn();
    render(
      <Button type={ButtonType.CLEAR} isDisabled={false} onClick={mockFn}>
        kuman514
      </Button>
    );

    fireEvent.click(screen.getByText(/kuman514/i));
    expect(mockFn).toBeCalledTimes(1);
  });

  it('should not be clickable on disabled', () => {
    const mockFn = vi.fn();
    render(
      <Button type={ButtonType.CLEAR} isDisabled onClick={mockFn}>
        kuman514
      </Button>
    );

    fireEvent.click(screen.getByText(/kuman514/i));
    expect(mockFn).toBeCalledTimes(0);
  });
});
