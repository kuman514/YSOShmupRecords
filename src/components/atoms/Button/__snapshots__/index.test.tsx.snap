// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sample test > should have a snapshot match for clear button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--primary-color);
  background-color: transparent;
  border: none;
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: var(--gray-color);
  background-color: transparent;
  text-decoration: line-through;
}

<button
  class="c0"
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for disabled clear button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--primary-color);
  background-color: transparent;
  border: none;
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: var(--gray-color);
  background-color: transparent;
  text-decoration: line-through;
}

<button
  class="c0"
  disabled=""
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for disabled line button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--primary-color);
  background-color: transparent;
  border: 1px solid var(--primary-color);
  border-radius: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
  disabled=""
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for disabled round line button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--primary-color);
  background-color: transparent;
  border: 1px solid var(--primary-color);
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
  disabled=""
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for disabled round solid button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: #ffffff;
  background-color: var(--primary-color);
  border: 1px solid var(--primary-color);
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
  disabled=""
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for disabled solid button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: #ffffff;
  background-color: var(--primary-color);
  border: 1px solid var(--primary-color);
  border-radius: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
  disabled=""
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for line button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--primary-color);
  background-color: transparent;
  border: 1px solid var(--primary-color);
  border-radius: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for round line button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--primary-color);
  background-color: transparent;
  border: 1px solid var(--primary-color);
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for round solid button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: #ffffff;
  background-color: var(--primary-color);
  border: 1px solid var(--primary-color);
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
>
  kuman514
</button>
`;

exports[`Sample test > should have a snapshot match for solid button 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-family: Inter;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: #ffffff;
  background-color: var(--primary-color);
  border: 1px solid var(--primary-color);
  border-radius: 4px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hover-color);
  color: #ffffff;
}

.c0:disabled {
  border: none;
  color: #ffffff;
  background-color: var(--gray-color);
  text-decoration: none;
}

<button
  class="c0"
>
  kuman514
</button>
`;
